CMakeLists.txt

```diff
@@ -41,8 +41,20 @@ project(Csound)
 
 enable_testing()
 
-set(libcsound_private_include_dirs "")
-set(libcsound_public_include_dirs "")
+# only libcsound
+add_library(libcsound_only INTERFACE)
+
+# for libcsound and plugins/utils that use internals
+add_library(libcsound_internals INTERFACE)
+
+# for users
+add_library(libcsound_interface INTERFACE)
+
+# for internal opcodes and plugins that use OpcodeBase.hpp
+add_library(opcode_internals INTERFACE)
+
+# for utils that use std_util.h and utilmain.h
+add_library(utility_internals INTERFACE)
```

Comment:
Less sure about this change. Why split them up so much?

Reply:
Hmm, well, none of them are used exclusively together, so there's no obvious way to combine them. Which ones would you want to combine? The `libcsound_only` exists almost solely for the `__BUILDING_LIBCSOUND` flag, otherwise I'd combine it with `libcsound_internals`

Reply:
I can't see any sources being added to this `utility_internals ` though no?

Reply:
Ref https://github.com/csound/csound/pull/1822#discussion_r1493382104

Reply:
I see a few advantages:

- You can see at a glance which parts of the interface are required for which targets (flags, include dirs, etc.)
- Targets don't get bloated with parts of the interface they don't need

Reply:
Eh, ok, I can see maybe this overcomplicates things. I combined stuff so now there's just two interfaces, `libcsound_internals` and `libcsound_interface`.